File: sdks/go/connectorkit/env.go
package connectorkit

import "os"

// RequireEnv returns env(name), or a *MissingEnvError when it is unset or empty.
//
// env is the replaceable seam docs/INCLUSION-POLICY.md §2 requires: a helper that reads
// the process environment with no way to override it fails criterion 2, which is exactly
// what TypeScript's requireProcessEnv does. Passing nil selects os.Getenv, so the common
// call is RequireEnv("API_TOKEN", nil) and the seam costs a caller nothing until they
// want it.
//
// func(string) string rather than Python's Mapping: os.Getenv already has this signature,
// so the stdlib supplies the default for free, and a read-only function gives a caller no
// seam that invites writing to the environment.
//
// An empty string counts as unset, matching Python and TypeScript.
func RequireEnv(name string, env func(string) string) (string, error) {
	if env == nil {
		env = os.Getenv
	}
	value := env(name)
	if value == "" {
		return "", &MissingEnvError{Name: name}
	}
	return value, nil
}

File: sdks/python/src/nimbus_sdk/connector_kit/env.py
"""Reading a required environment variable, through a replaceable seam.

TypeScript's ``requireProcessEnv`` reads ``process.env`` directly with no seam, which is
the exact pattern ``docs/INCLUSION-POLICY.md`` §2 names as a failure: *"a helper that
reads ``process.env.API_ENDPOINT`` with no way to override it still fails criterion 2."*
This binding is stricter than its original on purpose; the TypeScript fix is tracked as
a follow-up rather than replicated here for symmetry.
"""

from __future__ import annotations

import os
from collections.abc import Mapping

from nimbus_sdk.connector_kit.errors import MissingEnvError

__stability__ = "stable"


def require_env(name: str, env: Mapping[str, str] = os.environ) -> str:
    """Return ``env[name]``, or raise :class:`MissingEnvError` if unset or empty.

    ``env`` defaults to ``os.environ`` itself, not to a copy of it: the default is a
    live mapping, so a variable set after this module is imported is still visible. It
    is annotated as the read-only :class:`~collections.abc.Mapping` rather than
    ``MutableMapping`` — a helper whose job is reading the environment should not hand
    its caller a seam that invites writing to it.

    An empty string counts as unset, matching TypeScript's ``requireProcessEnv``.
    """
    value = env.get(name)
    if not value:
        raise MissingEnvError(f"{name} is not set")
    return value
